From 28b2acb0f02b648b053d3150de702c913bedd8e4 Mon Sep 17 00:00:00 2001
From: Robert Walker <bob.mt.wya@gmail.com>
Date: Wed, 29 May 2019 00:13:06 +1000
Subject: [PATCH 19/24] Infinality 2.10.0 (src/base/ftoutln.c)

---
 src/base/ftoutln.c                 |   10 +-
 1 file changed, 9 insertions(+), 1 deletions(-)

diff --git a/src/base/ftoutln.c b/src/base/ftoutln.c
index cb91321de..d5d12d682 100644
--- a/src/base/ftoutln.c
+++ b/src/base/ftoutln.c
@@ -22,7 +22,9 @@
 #include <freetype/internal/ftcalc.h>
 #include <freetype/internal/ftdebug.h>
 #include <freetype/fttrigon.h>
-
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+#include "ftinf.h"
+#endif
 
   /**************************************************************************
    *
@@ -900,7 +902,13 @@
     FT_Vector*      points;
     FT_Int          c, first, last;
     FT_Orientation  orientation;
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+    FT_Bool use_various_tweaks = FALSE;
+    if( ftinf ) use_various_tweaks=ftinf->use_various_tweaks;
 
+    if ( use_various_tweaks )
+      ystrength = FT_PIX_FLOOR ( ystrength );
+#endif
 
     if ( !outline )
       return FT_THROW( Invalid_Outline );
-- 
2.17.1

